test/rules/any.spec.js

```diff
@@ -9,7 +9,7 @@ describe("Test rule: any", () => {
 	it("should give back true anyway", () => {
 		const check = v.compile({ $$root: true, type: "any" });
 
-		expect(check(null)).toEqual([{ type: "required", actual: null, message: "The '' field is required." }]);
+		expect(check(null)).toEqual(true);
```

Comment:
It's not good that this case is changed. As you wrote in the PR name `null` is a value, so the `required` rule should accept it. It should be true if `nullable: false`

Reply:
Actually I'm not sure I'm following you on this point.
In my mind, the case seems good:
Here we consider that `nullable` is `true` by default as it's not explicitly set to false.
Given that `any` accept any type and with `$$root` modifier `any` accept any value, it can be anything but `undefined` which is not a value.
So when `any` type is used, `null` should be a valid value.
At the contrary, if `nullable=false` is set, `any` could be anything expect `null` or `undefined`.  
  
Or we will have to consider the default value of `nullable` to `false` but I'm scared that it will defeat the purpose and implies bigger changes to the test suite.

Reply:
Yeah, you are right, sorry I was wrong, I changed the lines in my mind. By the way, I would like to avoid the breaking change....breaking FV generates breaking changes in Moleculer, it means, we can't use it only just in the next version 0.15

To cover similar issue, we add a global option which switch the old/new behaviour. Can we do something similar here?

Reply:
No worries, I also had trouble getting my mind around it at first 😄 
I'll add to this PR a global setting to be able to switch between old and new behaviour.

Reply:
I've also added some documentation and tests that seems relevant for specific edge cases.

4 cases are tested when `considerNullAsAValue` option is set to `true`:
 - field is `required` and `nullable`, `default` value will **NOT** apply. (default)
 - field is `required` but **NOT** `nullable`, `default` value will apply if field is `null`.
 - field is `optional` and `nullable`, `default` will **NOT** apply.
 - field is `optional` but **NOT** `nullable`, `default` value will apply if field is `null`.

Reply:
Hi @icebob, feel free to tell me if you need me to do some other changes.

Reply:
Thanks, sorry for late, I'm checking...